src/test/javascript/spec/component/participation/participation.component.spec.ts

```diff
@@ -377,4 +379,48 @@ describe('ParticipationComponent', () => {
             expect(component.checkPresentationScoreConfig()).toBeFalse();
         });
     });
+
+    describe('getScoresRoute', () => {
+        const course1 = {
+            id: 1,
+            title: 'Course 1',
+        } as Course;
+
+        const exam1 = {
+            id: 100,
+            course: course1,
+        } as Exam;
+
+        const exerciseGroup1 = {
+            id: 50,
+            exam: exam1,
+        } as ExerciseGroup;
+
+        const exercise1 = {
+            id: 10,
+            title: 'Exercise 1',
+            type: 'text',
+            course: course1,
+        } as Exercise;
+
+        const exercise2 = {
+            id: 20,
+            title: 'Exercise 2',
+            type: 'programming',
+            exerciseGroup: exerciseGroup1,
+            course: undefined,
+        } as Exercise;
+
+        it('should return the correct route for an exercise without an exam', () => {
+            const expectedRoute = '/course-management/1/text-exercises/10/scores';
+            const result = component.getScoresRoute(exercise1);
+            expect(result).toEqual(expectedRoute);
+        });
+
+        it('should return the correct route for an exercise within an exam', () => {
+            const expectedRoute = '/course-management/1/exams/100/exercise-groups/50/programming-exercises/20/scores';
+            const result = component.getScoresRoute(exercise2);
+            expect(result).toEqual(expectedRoute);
+        });
```

Comment:
You can move the initialization of `course1` and `exercise1` to the first test and the initialization of `exercise2` to the second test, because these variables are used exclusively there.

Reply:
Will do!